### PR TITLE
Add stop and run button

### LINKED ISSUE #54: Add way to cancel/stop a build
Currently, i use reload window to cancel a build

I'm not sure if it is fine to just kill the child process

### PR DESCRIPTION
**Pull request by SeaDve**
Closes #54

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -131,6 +135,16 @@ export async function activate(context: ExtensionContext): Promise<void> {
       })
     )
 
+    // Stops
+    context.subscriptions.push(
+      registerCommand(`${EXT_ID}.stop`, () => {
+        if (store.state.getState().pipeline.application.built) {
```
Comment (bilelmoussaoui) [inline src/extension.ts:141]:
```suggestion
```
Doesn't sound correct, you might want to stop at it update deps for example

**Comment by bilelmoussaoui** [comment]
Do you mind also updating the CHANGELOG.md file with your recent changes along with rebasing this PR? :)

**Comment by SeaDve** [comment]
I'll do the changelog on a separate pr, so I won't mess up git hehe

**Comment by bilelmoussaoui** [comment]
> the stop method on FlatpakTerminal is currently broken since it does not stop every child process, only the current

The other child process are suppossed to be either not spawned yet or already successfully done. What is the issue you're facing here ?

**Comment by SeaDve** [comment]
The child process of the actual application doesn't get killed, causing subsequent rebuild to overlap running application

**Comment by bilelmoussaoui** [comment]
> The child process of the actual application doesn't get killed, causing subsequent rebuild and run to overlap

Not great, maybe we could check if the application itself is running (and it is not a build step like build dependencies) and run `flatpak kill $app-id` as well?

**Comment by SeaDve** [comment]
> run flatpak kill $app-id as well?

that fixes it

**Comment by SeaDve** [comment]
The UX/UI is not pretty good ATM, could be fixed though in #62

**Comment by bilelmoussaoui** [comment]
Ah that is very unfortunate

**Comment by bilelmoussaoui** [comment]
Which means you will have to manually restart rust-analyzer in this case I believe ?

**Comment by SeaDve** [comment]
that's also what i think. Is it run inside the child_process?

**Comment by SeaDve** [comment]
unfortunately, it doesn't improve the situation

**Comment by SeaDve** [comment]
Now working properly, without `flatpak kill` :D

It was not killing before because the other processes are shadowed by other processes because the commands are not being awaited

**Comment by bilelmoussaoui** [comment]
Ah I see, the code changes looks good to me. Although what I don't like is you need to at least do a build before being able to click on run because that will trigger rebuild :/ Maybe we could have a specific command for the "run" button that does either build or rebuild based on  the pipeline state

**Comment by SeaDve** [comment]
that could be handled in #63 and #73

**Pull request merged**